mmv1/third_party/terraform/services/compute/resource_compute_instance_group_manager_internal_test.go.erb

```diff
@@ -0,0 +1,298 @@
+		"projects/imre-test/global/instanceTemplates/example-template-custom?uniqueId=123":                                       "projects/imre-test/global/instanceTemplates/123",
+		"tf-test-igm-8amncgtq22?uniqueId=8361222501423044003":                                                                    "8361222501423044003",
+	}
+	for input, expected := range matches {
+		actual := ConvertToUniqueIdWhenPresent(input)
+		if actual != expected {
+			t.Fatalf("invalid return value by ConvertToUniqueIdWhenPresent for input %v; expected: %v, actual: %v", input, expected, actual)
+		}
+	}
+}
+
+func TestFlattenStatefulPolicyStatefulIps(t *testing.T) {
+	cases := map[string]struct {
+		ConfigValues []interface{}
+		Ips          map[string]compute.StatefulPolicyPreservedStateNetworkIp
+		Expected     []map[string]interface{}
+	}{
+		"No IPs in config nor API data": {
+			ConfigValues: []interface{}{},
+			Ips:          map[string]compute.StatefulPolicyPreservedStateNetworkIp{},
+			Expected:     []map[string]interface{}{},
+		},
+		"Single IP (nic0) in config and API data": {
+			ConfigValues: []interface{}{
+				map[string]interface{}{
+					"interface_name": "nic0",
+					"delete_rule":    "NEVER",
+				},
+			},
+			Ips: map[string]compute.StatefulPolicyPreservedStateNetworkIp{
+				"nic0": {
+					AutoDelete: "NEVER",
+				},
+			},
+			Expected: []map[string]interface{}{
+				{
+					"interface_name": "nic0",
+					"delete_rule":    "NEVER",
+				},
+			},
+		},
+		"Two IPs (nic0, nic1). Unordered in config and sorted in API data": {
+			ConfigValues: []interface{}{
+				map[string]interface{}{
+					"interface_name": "nic1",
+					"delete_rule":    "NEVER",
+				},
+				map[string]interface{}{
+					"interface_name": "nic0",
+					"delete_rule":    "NEVER",
+				},
+			},
+			Ips: map[string]compute.StatefulPolicyPreservedStateNetworkIp{
+				"nic0": {
+					AutoDelete: "NEVER",
+				},
+				"nic1": {
+					AutoDelete: "NEVER",
+				},
+			},
+			Expected: []map[string]interface{}{
+				{
+					"interface_name": "nic1",
+					"delete_rule":    "NEVER",
+				},
+				{
+					"interface_name": "nic0",
+					"delete_rule":    "NEVER",
+				},
+			},
+		},
+		"Two IPs (nic0, nic1). Only nic0 in config and both stored in API data": {
+			ConfigValues: []interface{}{
+				map[string]interface{}{
+					"interface_name": "nic0",
+					"delete_rule":    "NEVER",
+				},
+			},
+			Ips: map[string]compute.StatefulPolicyPreservedStateNetworkIp{
+				"nic0": {
+					AutoDelete: "NEVER",
+				},
+				"nic1": {
+					AutoDelete: "NEVER",
+				},
+			},
+			Expected: []map[string]interface{}{
+				{
+					"interface_name": "nic0",
+					"delete_rule":    "NEVER",
+				},
+				{
+					"interface_name": "nic1",
+					"delete_rule":    "NEVER",
+				},
+			},
+		},
+		"Two IPs (nic0, nic1). None stored in config and both stored in API data": {
```

Comment:
This test failed before because the data in the client library has uncertain order due to being a map's k:v pairs yet the provider converts that data into a slice and tries to preserve an order. The test case has 2 interfaces that aren't present in the config and the new sorting logic only acts to sort interfaces that match those in the config; interfaces not in the config are added in what ever order they come from the client library. If the client library has >1 of those then "the order they come from the client library" can vary due to randomness, which means tests cannot assert an expected order.

Other test cases in this unit test are fine because there's only ever 1 'unexpected' interface appended to the end of the slice.

One solution is to update this test case to only consider 1 interface, or another solution is to order the unexpected interfaces stored in [unexpectedIps](https://github.com/GoogleCloudPlatform/magic-modules/pull/9758/files#diff-f01f6698125e0a7cd08298c6a6e56df40709cad4caf8086f14c12760652d0c78R1342)

Reply:
I'll add ordering of the [unexpectedIps](https://github.com/GoogleCloudPlatform/magic-modules/pull/9758/files#diff-f01f6698125e0a7cd08298c6a6e56df40709cad4caf8086f14c12760652d0c78R1342) slice (that contains interfaces returned from the API that aren't in the config). This will allow this unit test to assert what data returned from the client library will look like without being affected by the randomness of converting maps to slices.

In a real world scenario, if a user has >1 interfaces that aren't present in their Terraform config they will see a diff in their plan and the order of those new fields might change in repeated plans. If I implemented ordering of `unexpectedIps` users will benefit by those new fields being displayed in a consistent order.